nsxt/resource_nsxt_policy_security_policy_no_rule.go

```diff
@@ -0,0 +1,149 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+
+	"github.com/vmware/terraform-provider-nsxt/api/infra/domains"
+)
+
+func resourceNsxtPolicySecurityPolicyNoRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicySecurityPolicyNoRuleCreate,
+		Read:   resourceNsxtPolicySecurityPolicyNoRuleRead,
+		Update: resourceNsxtPolicySecurityPolicyNoRuleUpdate,
+		Delete: resourceNsxtPolicySecurityPolicyNoRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: nsxtDomainResourceImporter,
+		},
+		Schema: getPolicySecurityPolicySchema(false, true, false),
+	}
+}
+
+func resourceNsxtPolicySecurityPolicyNoRuleCreate(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
```

Comment:
would it make sense to parametrize the CRUD to functions to either include or exclude the rule, in order to reuse the common code with the other policy resource?

Reply:
It could make sense but I think it might deserve a separate PR to do so, including switch from using H-API to using  security policy API for `nsxt_policy_security_policy`

Reply:
I just noticed that securityPolicy's API couldn't delete rules. So to let `parent_security_policy` and `security_policy` share the common code I changed to use H-API for the `parent_security_policy` with the latest commit. It will make more sense to conduct in this PR instead of another one.
And, as for the other policy resource, did you mean `gateway_policy` and `predefined_security_poilcy` as well? From my understanding, `predefined_security_policy` has a different schema than `security_policy`. And `gateway_policy`, although sharing most of the schema, is a different model. I'm not sure merging those two resources could share much common code. Seems making them self-contained is a better option.
Glad to hear your thoughts!

Reply:
Could you please clarify what you mean by lack of support for deleting rules from the API - I see [this](https://github.com/vmware/vsphere-automation-sdk-go/blob/master/services/nsxt/infra/domains/security_policies/RulesClient.go#L34) in the SDK. 
Regardless, follow up PR for optimizing's the code is fine by me!

Reply:
Sorry for the confusion. 
I mean I couldn't use the security policy [client](https://github.com/vmware/vsphere-automation-sdk-go/blob/master/services/nsxt/infra/domains/SecurityPoliciesClient.go) to delete the rule belonging to it. If we use the SDK for `Rule` to manage `security_policy`, not the parent one, we have to make several additional API calls for the rules, instead of one API call as H-API does. I'm a little bit concerned about the performance.

Reply:
I think when you fire the DELETE api call for the policy, all of its rules will be deleted - no need to delete them beforehand. What that your concern?

Reply:
I mean for example, there is a policy that has 200 rules and we want to delete 100 rules.